### DISCUSSION TITLE
Unexpected Models Behavior

### DISCUSSION THREAD

**Discussion by YaraAmin** [vladmandic/human#179] [Q&A] [answered]
I'm facing this issue with the models where I pass to them a static image and the results are different of the first detection from the second detection! for example, they can detect hands at one time and at another time on the same image they can't. Is there any explanation for that? Also, the post-processing output is different on the same image I believe this is from the different output from the models. Right?

**Comment by vladmandic** [comment]
if passing images, make sure that `config.cacheSensitivity = 0` to disable caching  

also, which version of `human`? hand model has been significantly changed in main branch tagged 2.3.0 recently and is still under development (thats why 2.3.0 hasnt been published)

**Comment by YaraAmin** [comment]
Thanks! can you explain to me how can we have caching while using a static image at a new session? or what do you mean exactly by cache sensitivity?

**Comment by YaraAmin** [comment]
Also how to install the new version "2.3.0" properly?

**Comment by vladmandic** [comment, marked as answer]
> can you explain to me how can we have caching while using a static image at a new session? or what do you mean exactly by cache sensitivity?

what do you mean by a new session?  

models are always cached for entire lifespan of human instance.

what `cacheSensitivity = 0.75` (which is default) means is "if frame changed less than 75% compared to previous frame, skip some processing and use cached results from previous frame"

for example, when detecting video input, we don't need to re-run hand detection on each frame since we know approximate location of the hand from previous frame - so we can only run hand analysis using best-guess hand location.

when running image analysis, makes no sense to cache results. and human tries to guess that, but sometimes its best to set `cacheSensitivity = 0` explicitly so it doesn't have to analyze before/after at all.

> Also how to install the new version "2.3.0" properly?

you can install package from git main branch: `npm install git+ssh://github.com/vladmandic/human.git`

**Comment by YaraAmin** [comment]
Thank you so much!
